pkg/cmd/cli/describe/projectstatus.go

```diff
@@ -326,38 +339,101 @@ func indentLines(indent string, lines ...string) []string {
 	return ret
 }
 
-func describeProjectAndServer(project *projectapi.Project, server string) string {
-	if server != "" {
+type formatter interface {
+	ResourceName(obj interface{}) string
+}
+
+func namespaceNameWithType(resource, name, namespace, defaultNamespace string, noNamespace bool) string {
+	if noNamespace || namespace == defaultNamespace || len(namespace) == 0 {
+		return resource + "/" + name
+	}
+	return resource + "/" + name + "[" + namespace + "]"
```

Comment:
It looks horribly ugly in the output separated.  If we didn't do the
rc/blah it might read better, but that just means the line widths explode.
Parenthesis and most other symbols didn't improve it.

On Dec 18, 2015, at 4:34 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/cmd/cli/describe/projectstatus.go
<https://github.com/openshift/origin/pull/6389#discussion_r48072464>:

> @@ -326,38 +339,101 @@ func indentLines(indent string, lines ...string) []string {
>  	return ret
>  }
>
> -func describeProjectAndServer(project *projectapi.Project, server string) string {
> -	if server != "" {
> +type formatter interface {
> +	ResourceName(obj interface{}) string
> +}
> +
> +func namespaceNameWithType(resource, name, namespace, defaultNamespace string, noNamespace bool) string {
> +	if noNamespace || namespace == defaultNamespace || len(namespace) == 0 {
> +		return resource + "/" + name
> +	}
> +	return resource + "/" + name + "[" + namespace + "]"

I still prefer separating the namespace completely... it feels weird to
group it with either the name or the resource

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/6389/files#r48072464>.